manifests/config.pp

```diff
@@ -441,4 +443,14 @@
     # make sure service gets restarted after systemctl daemon-reload
     Class['systemd::systemctl::daemon_reload'] ~> Class['rsyslog::service']
   }
+
+# give deprecation warning if rsyslog is 8.6 or later and the -l or -s options
+# are being used.
+  if $facts['rsyslogd'] and $facts['rsyslogd']['version'] {
+    if versioncmp($facts['rsyslogd']['version'], '8.6.0') > 0  {
+      if ! empty($host_list) or ! empty($domain_list) {
+        warning('Rsyslog has deprecated the -l and -s options in version 8.6.0 and later')
```

Comment:
Would be nice if this message also indicated the fully-qualified name of OBE parameters that have set